src/ts/core/features/filter-search.ts

```diff
@@ -0,0 +1,130 @@
+    {
+        if(debugMode == 1){console.log(evtTarget.className);}
+
+        if(evtTarget.className === 'bp3-button')
+        {
+            var tbInputSearch = (<HTMLInputElement>document.getElementById("fbSearchInput"));
+            if(tbInputSearch !== null)
+            {
+                tbInputSearch.focus();
+                tbInputSearch.select();
+            }
+        }
+        else if(evtTarget.className === 'bp3-icon bp3-icon-filter' || evtTarget.className === 'bp3-button bp3-minimal bp3-small')
+        {
+            var waitCtr = 0;
+            function waitForFilter()
+            {
+                setTimeout(function(){
+                    //Check if filter search input box is there, otherwise need to load it
+                    var tbInputSearch = (<HTMLInputElement>document.getElementById("fbSearchInput"));
+
+                    if(tbInputSearch === null)
+                    {
+                        var allByClassTest = document.querySelectorAll('div.bp3-transition-container.bp3-popover-enter-done div.bp3-popover-content > div > div');
+                        var testFilterDiv = (<HTMLElement>allByClassTest[0]);
+                        if(typeof testFilterDiv !== 'undefined' && testFilterDiv !== null)
+                        {
+                            var newDivLine = document.createElement('div');
+                                newDivLine.className = 'rm-line';
+                                testFilterDiv.prepend(newDivLine);
+
+                            var newDiv = document.createElement('div');
+                                newDiv.id = 'filterBoxSearch';
+                                newDiv.style.cssText = 'display:flex;padding:4px';
+                                testFilterDiv.prepend(newDiv);
+
+                            var newLabel = document.createElement('strong');
+                                newLabel.innerText = 'Search';
+                                newLabel.style.cssText = 'margin-right:10px';
+                                newDiv.appendChild(newLabel);
+
+                            var newInput = document.createElement('input');
+                                newInput.value = '';
+                                newInput.id = 'fbSearchInput';
+                                newInput.name = 'fbSearchInput';
+                                newInput.style.cssText = 'width:200px;display:flex';
+                                newDiv.appendChild(newInput);
+                                newInput.focus();
+
+                            newInput.addEventListener("input", newInputClick);
+                            return;
+                        }else{if(debugMode == 1){console.log('Filter box was not loaded in time!')}}
+                    }
+                    waitCtr++;
+                    if(debugMode == 1){console.log('*** WAIT LOOP: ',waitCtr)}
+                    if(waitCtr >= 12){return;}
+                    waitForFilter();
+                }, 50)
```

Comment:
this seems like you are waiting for element to be created, and then want to execute some action on it, right? https://github.com/uzairfarooq/arrive seems like something that would allow us do things like this nicer (you can add it to the project by running `npm install arrive --save`

Reply:
ok, updated recommendation. common utils now has https://github.com/roam-unofficial/roam-toolkit/blob/master/src/ts/core/common/mutation-observer.ts#L26 which should work here